r/aws_appmesh_route,r/aws_appmesh_virtual_node,r/aws_appmesh_virtual_router: Support gRPC and HTTP/2 services

ewbankkit opened the pull request:
<!--- See what makes a good Pull Request at : https://github.com/terraform-providers/terraform-provider-aws/blob/master/.github/CONTRIBUTING.md#pull-requests --->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

<!--- Thank you for keeping this note for the community --->

<!--- If your PR fully resolves and should automatically close the linked issue, use Closes. Otherwise, use Relates --->
Closes https://github.com/terraform-providers/terraform-provider-aws/issues/10803.
Relates: https://github.com/terraform-providers/terraform-provider-aws/issues/14601.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):
<!--
If change is not user facing, just write "NONE" in the release-note block below.
-->

```release-note
resource/aws_appmesh_route: Add `grpc_route` and `http2_route` attributes to support gRPC and HTTP/2 services
resource/aws_appmesh_virtual_node: Add `grpc` and `http2` as valid values for the `protocol` attribute
resource/aws_appmesh_virtual_router: Add `grpc` and `http2` as valid values for the `protocol` attribute
```

Output from acceptance testing:

<!--
Replace TestAccXXX with a pattern that matches the tests affected by this PR.

For more information on the `-run` flag, see the `go test` documentation at https://tip.golang.org/cmd/go/#hdr-Testing_flags.
-->
```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSAppmesh/Route'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAppmesh/Route -timeout 120m
=== RUN   TestAccAWSAppmesh
=== RUN   TestAccAWSAppmesh/Route
=== RUN   TestAccAWSAppmesh/Route/routePriority
=== RUN   TestAccAWSAppmesh/Route/tcpRoute
=== RUN   TestAccAWSAppmesh/Route/tags
=== RUN   TestAccAWSAppmesh/Route/grpcRoute
=== RUN   TestAccAWSAppmesh/Route/http2Route
=== RUN   TestAccAWSAppmesh/Route/httpHeader
=== RUN   TestAccAWSAppmesh/Route/httpRetryPolicy
=== RUN   TestAccAWSAppmesh/Route/httpRoute
=== RUN   TestAccAWSAppmesh/VirtualRouter
=== RUN   TestAccAWSAppmesh/VirtualRouter/basic
=== RUN   TestAccAWSAppmesh/VirtualRouter/tags
--- PASS: TestAccAWSAppmesh (539.34s)
    --- PASS: TestAccAWSAppmesh/Route (429.82s)
        --- PASS: TestAccAWSAppmesh/Route/routePriority (51.69s)
        --- PASS: TestAccAWSAppmesh/Route/tcpRoute (49.77s)
        --- PASS: TestAccAWSAppmesh/Route/tags (73.48s)
        --- PASS: TestAccAWSAppmesh/Route/grpcRoute (50.72s)
        --- PASS: TestAccAWSAppmesh/Route/http2Route (50.98s)
        --- PASS: TestAccAWSAppmesh/Route/httpHeader (51.27s)
        --- PASS: TestAccAWSAppmesh/Route/httpRetryPolicy (50.96s)
        --- PASS: TestAccAWSAppmesh/Route/httpRoute (50.94s)
    --- PASS: TestAccAWSAppmesh/VirtualRouter (109.53s)
        --- PASS: TestAccAWSAppmesh/VirtualRouter/basic (45.55s)
        --- PASS: TestAccAWSAppmesh/VirtualRouter/tags (63.98s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	539.379s
```

Includes changes from ~#10402 and~ https://github.com/terraform-providers/terraform-provider-aws/pull/11660.

ewbankkit commented:
Removed _WIP_.
Ready for review.

ewbankkit commented:
Rebased to remove merge conflicts.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSAppmesh/Route'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAppmesh/Route -timeout 120m
=== RUN   TestAccAWSAppmesh
=== RUN   TestAccAWSAppmesh/Route
=== RUN   TestAccAWSAppmesh/Route/httpRetryPolicy
=== RUN   TestAccAWSAppmesh/Route/httpRoute
=== RUN   TestAccAWSAppmesh/Route/routePriority
=== RUN   TestAccAWSAppmesh/Route/tcpRoute
=== RUN   TestAccAWSAppmesh/Route/tags
=== RUN   TestAccAWSAppmesh/Route/grpcRoute
=== RUN   TestAccAWSAppmesh/Route/http2Route
=== RUN   TestAccAWSAppmesh/Route/httpHeader
=== RUN   TestAccAWSAppmesh/VirtualRouter
=== RUN   TestAccAWSAppmesh/VirtualRouter/basic
=== RUN   TestAccAWSAppmesh/VirtualRouter/tags
--- PASS: TestAccAWSAppmesh (536.64s)
    --- PASS: TestAccAWSAppmesh/Route (427.21s)
        --- PASS: TestAccAWSAppmesh/Route/httpRetryPolicy (51.00s)
        --- PASS: TestAccAWSAppmesh/Route/httpRoute (50.84s)
        --- PASS: TestAccAWSAppmesh/Route/routePriority (50.68s)
        --- PASS: TestAccAWSAppmesh/Route/tcpRoute (50.89s)
        --- PASS: TestAccAWSAppmesh/Route/tags (71.32s)
        --- PASS: TestAccAWSAppmesh/Route/grpcRoute (50.65s)
        --- PASS: TestAccAWSAppmesh/Route/http2Route (50.99s)
        --- PASS: TestAccAWSAppmesh/Route/httpHeader (50.84s)
    --- PASS: TestAccAWSAppmesh/VirtualRouter (109.44s)
        --- PASS: TestAccAWSAppmesh/VirtualRouter/basic (45.38s)
        --- PASS: TestAccAWSAppmesh/VirtualRouter/tags (64.05s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	536.686s
```

ewbankkit commented:
Rebased.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSAppmesh/Route/'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAppmesh/Route/ -timeout 120m
=== RUN   TestAccAWSAppmesh
=== RUN   TestAccAWSAppmesh/Route
=== RUN   TestAccAWSAppmesh/Route/tags
=== RUN   TestAccAWSAppmesh/Route/grpcRoute
=== RUN   TestAccAWSAppmesh/Route/http2Route
=== RUN   TestAccAWSAppmesh/Route/httpHeader
=== RUN   TestAccAWSAppmesh/Route/httpRetryPolicy
=== RUN   TestAccAWSAppmesh/Route/httpRoute
=== RUN   TestAccAWSAppmesh/Route/routePriority
=== RUN   TestAccAWSAppmesh/Route/tcpRoute
=== RUN   TestAccAWSAppmesh/VirtualRouter
=== RUN   TestAccAWSAppmesh/VirtualRouter/basic
=== RUN   TestAccAWSAppmesh/VirtualRouter/tags
--- PASS: TestAccAWSAppmesh (537.75s)
    --- PASS: TestAccAWSAppmesh/Route (428.52s)
        --- PASS: TestAccAWSAppmesh/Route/tags (72.36s)
        --- PASS: TestAccAWSAppmesh/Route/grpcRoute (51.04s)
        --- PASS: TestAccAWSAppmesh/Route/http2Route (51.50s)
        --- PASS: TestAccAWSAppmesh/Route/httpHeader (51.06s)
        --- PASS: TestAccAWSAppmesh/Route/httpRetryPolicy (51.69s)
        --- PASS: TestAccAWSAppmesh/Route/httpRoute (51.24s)
        --- PASS: TestAccAWSAppmesh/Route/routePriority (49.93s)
        --- PASS: TestAccAWSAppmesh/Route/tcpRoute (49.70s)
    --- PASS: TestAccAWSAppmesh/VirtualRouter (109.23s)
        --- PASS: TestAccAWSAppmesh/VirtualRouter/basic (45.68s)
        --- PASS: TestAccAWSAppmesh/VirtualRouter/tags (63.55s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	537.783s
```

AdamDM commented:
I couldn't get this working until I added the changes from this to it https://github.com/terraform-providers/terraform-provider-aws/commit/ece07d65179e4b9591de312a1b43f2cd197323ba

ewbankkit commented:
@AdamDM Thanks for that; I'll merge https://github.com/terraform-providers/terraform-provider-aws/pull/12744 into this PR if the author is OK with that.

stijndehaes commented:
> @AdamDM Thanks; I'll merge #12744 into this PR if the author is OK with that.

go ahead!

ewbankkit commented:
Incorporated changes from https://github.com/terraform-providers/terraform-provider-aws/pull/12744 and modified tests for `aws_appmesh_virtual_router` gRPC and HTTP/2 protocols:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSAppmesh/Route'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAppmesh/Route -timeout 120m
=== RUN   TestAccAWSAppmesh
=== RUN   TestAccAWSAppmesh/VirtualRouter
=== RUN   TestAccAWSAppmesh/VirtualRouter/tags
=== RUN   TestAccAWSAppmesh/VirtualRouter/basic
=== RUN   TestAccAWSAppmesh/Route
=== RUN   TestAccAWSAppmesh/Route/routePriority
=== RUN   TestAccAWSAppmesh/Route/tcpRoute
=== RUN   TestAccAWSAppmesh/Route/tags
=== RUN   TestAccAWSAppmesh/Route/grpcRoute
=== RUN   TestAccAWSAppmesh/Route/http2Route
=== RUN   TestAccAWSAppmesh/Route/httpHeader
=== RUN   TestAccAWSAppmesh/Route/httpRetryPolicy
=== RUN   TestAccAWSAppmesh/Route/httpRoute
--- PASS: TestAccAWSAppmesh (536.98s)
    --- PASS: TestAccAWSAppmesh/VirtualRouter (109.52s)
        --- PASS: TestAccAWSAppmesh/VirtualRouter/tags (64.84s)
        --- PASS: TestAccAWSAppmesh/VirtualRouter/basic (44.67s)
    --- PASS: TestAccAWSAppmesh/Route (427.47s)
        --- PASS: TestAccAWSAppmesh/Route/routePriority (50.77s)
        --- PASS: TestAccAWSAppmesh/Route/tcpRoute (50.70s)
        --- PASS: TestAccAWSAppmesh/Route/tags (71.37s)
        --- PASS: TestAccAWSAppmesh/Route/grpcRoute (50.81s)
        --- PASS: TestAccAWSAppmesh/Route/http2Route (51.67s)
        --- PASS: TestAccAWSAppmesh/Route/httpHeader (50.53s)
        --- PASS: TestAccAWSAppmesh/Route/httpRetryPolicy (51.11s)
        --- PASS: TestAccAWSAppmesh/Route/httpRoute (50.50s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	537.033s
```

ewbankkit commented:
Revisit, considering #13194.

ewbankkit commented:
Rebase to remove merge conflicts.

Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAppmesh/VirtualNode/'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAppmesh/VirtualNode/ -timeout 120m
=== RUN   TestAccAWSAppmesh
=== RUN   TestAccAWSAppmesh/VirtualNode
=== RUN   TestAccAWSAppmesh/VirtualNode/basic
=== RUN   TestAccAWSAppmesh/VirtualNode/cloudMapServiceDiscovery
=== RUN   TestAccAWSAppmesh/VirtualNode/listenerHealthChecks
=== RUN   TestAccAWSAppmesh/VirtualNode/logging
=== RUN   TestAccAWSAppmesh/VirtualNode/tags
--- PASS: TestAccAWSAppmesh (301.77s)
    --- PASS: TestAccAWSAppmesh/VirtualNode (301.77s)
        --- PASS: TestAccAWSAppmesh/VirtualNode/basic (28.09s)
        --- PASS: TestAccAWSAppmesh/VirtualNode/cloudMapServiceDiscovery (117.55s)
        --- PASS: TestAccAWSAppmesh/VirtualNode/listenerHealthChecks (45.81s)
        --- PASS: TestAccAWSAppmesh/VirtualNode/logging (45.58s)
        --- PASS: TestAccAWSAppmesh/VirtualNode/tags (64.75s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	301.825s
```

ewbankkit commented:
Rebased to remove merge conflicts.
Use new `TypeSet` acceptance check tests routines - #13556.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAppmesh/Route/'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAppmesh/Route/ -timeout 120m
=== RUN   TestAccAWSAppmesh
=== RUN   TestAccAWSAppmesh/Route
=== RUN   TestAccAWSAppmesh/Route/httpHeader
=== RUN   TestAccAWSAppmesh/Route/httpRetryPolicy
=== RUN   TestAccAWSAppmesh/Route/httpRoute
=== RUN   TestAccAWSAppmesh/Route/routePriority
=== RUN   TestAccAWSAppmesh/Route/tcpRoute
=== RUN   TestAccAWSAppmesh/Route/tags
=== RUN   TestAccAWSAppmesh/Route/grpcRoute
=== RUN   TestAccAWSAppmesh/Route/http2Route
=== RUN   TestAccAWSAppmesh/VirtualRouter
=== RUN   TestAccAWSAppmesh/VirtualRouter/basic
=== RUN   TestAccAWSAppmesh/VirtualRouter/tags
--- PASS: TestAccAWSAppmesh (609.10s)
    --- PASS: TestAccAWSAppmesh/Route (497.03s)
        --- PASS: TestAccAWSAppmesh/Route/httpHeader (51.51s)
        --- PASS: TestAccAWSAppmesh/Route/httpRetryPolicy (51.39s)
        --- PASS: TestAccAWSAppmesh/Route/httpRoute (71.47s)
        --- PASS: TestAccAWSAppmesh/Route/routePriority (52.58s)
        --- PASS: TestAccAWSAppmesh/Route/tcpRoute (72.59s)
        --- PASS: TestAccAWSAppmesh/Route/tags (73.42s)
        --- PASS: TestAccAWSAppmesh/Route/grpcRoute (72.79s)
        --- PASS: TestAccAWSAppmesh/Route/http2Route (51.28s)
    --- PASS: TestAccAWSAppmesh/VirtualRouter (112.07s)
        --- PASS: TestAccAWSAppmesh/VirtualRouter/basic (46.21s)
        --- PASS: TestAccAWSAppmesh/VirtualRouter/tags (65.86s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	609.171s
```

ewbankkit commented:
Rebased and re-ran acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSAppmesh'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAppmesh -timeout 120m
=== RUN   TestAccAWSAppmesh_serial
=== RUN   TestAccAWSAppmesh_serial/VirtualService
=== RUN   TestAccAWSAppmesh_serial/VirtualService/virtualNode
=== RUN   TestAccAWSAppmesh_serial/VirtualService/virtualRouter
=== RUN   TestAccAWSAppmesh_serial/VirtualService/tags
=== RUN   TestAccAWSAppmesh_serial/Mesh
=== RUN   TestAccAWSAppmesh_serial/Mesh/basic
=== RUN   TestAccAWSAppmesh_serial/Mesh/egressFilter
=== RUN   TestAccAWSAppmesh_serial/Mesh/tags
=== RUN   TestAccAWSAppmesh_serial/Route
=== RUN   TestAccAWSAppmesh_serial/Route/tcpRoute
=== RUN   TestAccAWSAppmesh_serial/Route/tags
=== RUN   TestAccAWSAppmesh_serial/Route/grpcRoute
=== RUN   TestAccAWSAppmesh_serial/Route/http2Route
=== RUN   TestAccAWSAppmesh_serial/Route/httpHeader
=== RUN   TestAccAWSAppmesh_serial/Route/httpRetryPolicy
=== RUN   TestAccAWSAppmesh_serial/Route/httpRoute
=== RUN   TestAccAWSAppmesh_serial/Route/routePriority
=== RUN   TestAccAWSAppmesh_serial/VirtualNode
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/basic
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/cloudMapServiceDiscovery
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/listenerHealthChecks
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/logging
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/tags
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter/basic
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter/tags
--- PASS: TestAccAWSAppmesh_serial (658.25s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualService (91.92s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/virtualNode (28.05s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/virtualRouter (25.79s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/tags (38.07s)
    --- PASS: TestAccAWSAppmesh_serial/Mesh (57.71s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/basic (10.84s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/egressFilter (22.34s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/tags (24.53s)
    --- PASS: TestAccAWSAppmesh_serial/Route (266.76s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tcpRoute (38.11s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tags (39.15s)
        --- PASS: TestAccAWSAppmesh_serial/Route/grpcRoute (38.63s)
        --- PASS: TestAccAWSAppmesh_serial/Route/http2Route (28.12s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpHeader (28.02s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRetryPolicy (28.02s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRoute (38.48s)
        --- PASS: TestAccAWSAppmesh_serial/Route/routePriority (28.22s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualNode (187.33s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/basic (14.57s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/cloudMapServiceDiscovery (94.98s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/listenerHealthChecks (22.92s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/logging (22.73s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/tags (32.11s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualRouter (54.54s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/basic (22.71s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/tags (31.83s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	658.337s
```

ewbankkit commented:
Rebased and migrated acceptance test and documentation syntax to TF 0.12.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAppmesh/Route/'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAppmesh/Route/ -timeout 120m
=== RUN   TestAccAWSAppmesh_serial
=== RUN   TestAccAWSAppmesh_serial/Route
=== RUN   TestAccAWSAppmesh_serial/Route/tcpRoute
=== RUN   TestAccAWSAppmesh_serial/Route/tags
=== RUN   TestAccAWSAppmesh_serial/Route/grpcRoute
=== RUN   TestAccAWSAppmesh_serial/Route/http2Route
=== RUN   TestAccAWSAppmesh_serial/Route/httpHeader
=== RUN   TestAccAWSAppmesh_serial/Route/httpRetryPolicy
=== RUN   TestAccAWSAppmesh_serial/Route/httpRoute
=== RUN   TestAccAWSAppmesh_serial/Route/routePriority
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter/basic
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter/tags
--- PASS: TestAccAWSAppmesh_serial (445.49s)
    --- PASS: TestAccAWSAppmesh_serial/Route (365.54s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tcpRoute (53.92s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tags (54.25s)
        --- PASS: TestAccAWSAppmesh_serial/Route/grpcRoute (53.66s)
        --- PASS: TestAccAWSAppmesh_serial/Route/http2Route (37.42s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpHeader (37.67s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRetryPolicy (37.44s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRoute (53.63s)
        --- PASS: TestAccAWSAppmesh_serial/Route/routePriority (37.54s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualRouter (79.94s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/basic (32.84s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/tags (47.11s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	445.526s
```

ewbankkit commented:
Rebased and fixed merge conflicts.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSAppmesh'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAppmesh -timeout 120m
=== RUN   TestAccAWSAppmesh_serial
=== RUN   TestAccAWSAppmesh_serial/Route
=== RUN   TestAccAWSAppmesh_serial/Route/httpRoute
=== RUN   TestAccAWSAppmesh_serial/Route/routePriority
=== RUN   TestAccAWSAppmesh_serial/Route/tcpRoute
=== RUN   TestAccAWSAppmesh_serial/Route/tags
=== RUN   TestAccAWSAppmesh_serial/Route/grpcRoute
=== RUN   TestAccAWSAppmesh_serial/Route/http2Route
=== RUN   TestAccAWSAppmesh_serial/Route/httpHeader
=== RUN   TestAccAWSAppmesh_serial/Route/httpRetryPolicy
=== RUN   TestAccAWSAppmesh_serial/VirtualNode
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/tags
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/basic
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/cloudMapServiceDiscovery
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/listenerHealthChecks
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/logging
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter/basic
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter/tags
=== RUN   TestAccAWSAppmesh_serial/VirtualService
=== RUN   TestAccAWSAppmesh_serial/VirtualService/virtualNode
=== RUN   TestAccAWSAppmesh_serial/VirtualService/virtualRouter
=== RUN   TestAccAWSAppmesh_serial/VirtualService/tags
=== RUN   TestAccAWSAppmesh_serial/Mesh
=== RUN   TestAccAWSAppmesh_serial/Mesh/egressFilter
=== RUN   TestAccAWSAppmesh_serial/Mesh/tags
=== RUN   TestAccAWSAppmesh_serial/Mesh/basic
--- PASS: TestAccAWSAppmesh_serial (928.20s)
    --- PASS: TestAccAWSAppmesh_serial/Route (374.85s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRoute (55.29s)
        --- PASS: TestAccAWSAppmesh_serial/Route/routePriority (38.22s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tcpRoute (54.46s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tags (55.72s)
        --- PASS: TestAccAWSAppmesh_serial/Route/grpcRoute (54.41s)
        --- PASS: TestAccAWSAppmesh_serial/Route/http2Route (39.10s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpHeader (39.54s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRetryPolicy (38.12s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualNode (244.71s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/tags (48.78s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/basic (21.49s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/cloudMapServiceDiscovery (105.94s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/listenerHealthChecks (34.63s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/logging (33.89s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualRouter (82.30s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/basic (34.00s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/tags (48.31s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualService (128.49s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/virtualNode (39.65s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/virtualRouter (34.58s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/tags (54.26s)
    --- PASS: TestAccAWSAppmesh_serial/Mesh (97.85s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/egressFilter (38.36s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/tags (42.28s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/basic (17.21s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	928.243s
```

ewbankkit commented:
Rebased and fixed merge conflicts.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSAppmesh'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAppmesh -timeout 120m
=== RUN   TestAccAWSAppmesh_serial
=== RUN   TestAccAWSAppmesh_serial/Mesh
=== RUN   TestAccAWSAppmesh_serial/Mesh/basic
=== RUN   TestAccAWSAppmesh_serial/Mesh/egressFilter
=== RUN   TestAccAWSAppmesh_serial/Mesh/tags
=== RUN   TestAccAWSAppmesh_serial/Route
=== RUN   TestAccAWSAppmesh_serial/Route/tcpRoute
=== RUN   TestAccAWSAppmesh_serial/Route/tags
=== RUN   TestAccAWSAppmesh_serial/Route/grpcRoute
=== RUN   TestAccAWSAppmesh_serial/Route/http2Route
=== RUN   TestAccAWSAppmesh_serial/Route/httpHeader
=== RUN   TestAccAWSAppmesh_serial/Route/httpRetryPolicy
=== RUN   TestAccAWSAppmesh_serial/Route/httpRoute
=== RUN   TestAccAWSAppmesh_serial/Route/routePriority
=== RUN   TestAccAWSAppmesh_serial/VirtualNode
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/backendDefaults
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/clientPolicyAcm
    resource_aws_appmesh_virtual_node_test.go:597: [INFO] Got non-empty plan, as expected
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/listenerHealthChecks
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/logging
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/tags
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/basic
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/clientPolicyFile
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/cloudMapServiceDiscovery
=== RUN   TestAccAWSAppmesh_serial/VirtualNode/tls
    resource_aws_appmesh_virtual_node_test.go:389: [INFO] Got non-empty plan, as expected
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter/basic
=== RUN   TestAccAWSAppmesh_serial/VirtualRouter/tags
=== RUN   TestAccAWSAppmesh_serial/VirtualService
=== RUN   TestAccAWSAppmesh_serial/VirtualService/virtualNode
=== RUN   TestAccAWSAppmesh_serial/VirtualService/virtualRouter
=== RUN   TestAccAWSAppmesh_serial/VirtualService/tags
--- PASS: TestAccAWSAppmesh_serial (1156.81s)
    --- PASS: TestAccAWSAppmesh_serial/Mesh (98.18s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/basic (17.66s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/egressFilter (38.11s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/tags (42.41s)
    --- PASS: TestAccAWSAppmesh_serial/Route (380.74s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tcpRoute (55.65s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tags (57.21s)
        --- PASS: TestAccAWSAppmesh_serial/Route/grpcRoute (56.46s)
        --- PASS: TestAccAWSAppmesh_serial/Route/http2Route (39.04s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpHeader (39.05s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRetryPolicy (39.11s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRoute (55.02s)
        --- PASS: TestAccAWSAppmesh_serial/Route/routePriority (39.19s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualNode (467.13s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/backendDefaults (34.01s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/clientPolicyAcm (61.38s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/listenerHealthChecks (34.46s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/logging (34.85s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/tags (49.70s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/basic (19.70s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/clientPolicyFile (34.36s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/cloudMapServiceDiscovery (106.55s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/tls (92.11s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualRouter (82.12s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/basic (33.36s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/tags (48.76s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualService (128.64s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/virtualNode (38.92s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/virtualRouter (34.40s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/tags (55.32s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1156.854s
```

DrFaust92 reviewed (changes requested):
```
-- FAIL: TestAccAWSAppmesh_serial (2467.58s)
    --- PASS: TestAccAWSAppmesh_serial/Mesh (224.73s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/basic (37.86s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/egressFilter (84.46s)
        --- PASS: TestAccAWSAppmesh_serial/Mesh/tags (102.41s)
    --- FAIL: TestAccAWSAppmesh_serial/Route (909.10s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tags (120.83s)
        --- PASS: TestAccAWSAppmesh_serial/Route/grpcRoute (123.10s)
        --- PASS: TestAccAWSAppmesh_serial/Route/http2Route (83.29s)
        --- FAIL: TestAccAWSAppmesh_serial/Route/httpHeader (178.07s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRetryPolicy (84.45s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpRoute (116.27s)
        --- PASS: TestAccAWSAppmesh_serial/Route/routePriority (84.94s)
        --- PASS: TestAccAWSAppmesh_serial/Route/tcpRoute (118.15s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualNode (880.74s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/basic (44.14s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/clientPolicyAcm (118.09s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/cloudMapServiceDiscovery (156.77s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/logging (73.40s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/backendDefaults (71.28s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/clientPolicyFile (71.14s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/listenerHealthChecks (72.02s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/tls (164.08s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualNode/tags (109.82s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualRouter (171.58s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/basic (69.37s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualRouter/tags (102.20s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualService (281.44s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/virtualNode (83.29s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/virtualRouter (79.41s)
        --- PASS: TestAccAWSAppmesh_serial/VirtualService/tags (118.74s)

    --- PASS: TestAccAWSAppmesh_serial/Route (92.54s)
        --- PASS: TestAccAWSAppmesh_serial/Route/httpHeader (92.54s)
    --- PASS: TestAccAWSAppmesh_serial/VirtualRouter (0.00s)
PASS
```

DrFaust92 reviewed (approved):
LGTM
